### ISSUE TITLE
Support Android 15

### ISSUE THREAD

**Issue by kelson42** [kiwix/kiwix-android#3990]
Android 15 is going to be released soon according to Google
https://www.androidcentral.com/apps-software/android-15

Can we fully support it? What would that take?

**Comment by MohitMaliFtechiz** [comment]
@kelson42 Yes we can add support for the Android 15 in our app. To fully support the Android 15 we have to test with the "private storage" like a ticket https://github.com/kiwix/kiwix-android/issues/3989.
Also in Android 15, there are some changes in foreground services. So we must adjust our "Read Aloud", and "Hotspot" services.
Also, the new gradle that supports the Android 15 will come with some improvements and deprecation of methods so we have to refactored our code according to it.

These some points on which we have to work while adding the support for Android 15.
